mp2-common/src/eth.rs

```diff
@@ -111,6 +116,175 @@ pub struct ProofQuery {
+    /// Packed event signature,
+    pub event_signature: [u8; 32],
+    /// Byte offset from the start of the log to event signature
+    pub sig_rel_offset: usize,
+    /// The topics for this Log
+    pub topics: [LogDataInfo; 3],
+    /// The extra data stored by this Log
+    pub data: [LogDataInfo; 2],
+}
+
+/// Contains all the information for data contained in an [`Event`]
+#[derive(Debug, Clone, Copy, Default, Serialize, Deserialize)]
+pub struct LogDataInfo {
+    pub column_id: usize,
+    /// The byte offset from the beggining of the log to this target
+    pub rel_byte_offset: usize,
+    /// The length of this topic/data
+    pub len: usize,
+}
+
+impl TryFrom<&Log<LogData>> for EventLogInfo {
+    type Error = anyhow::Error;
+
+    fn try_from(log: &Log<LogData>) -> std::result::Result<Self, Self::Error> {
+        // First we encode the log in rlp form
+        let mut buf = Vec::<u8>::new();
+        log.encode(&mut buf);
+
+        let rlp_log = rlp::Rlp::new(&buf);
+        // Extract the header
+        let log_header = rlp_log.payload_info()?;
+        let next_data = &buf[log_header.header_len..log_header.header_len + log_header.value_len];
+        let rlp_log_no_header = rlp::Rlp::new(next_data);
+        // Find the address offset (skipping its header)
+        let address_header = rlp_log_no_header.payload_info()?;
+        let rel_address_offset = log_header.header_len + address_header.header_len;
+        // Find the signature offset (skipping its header)
+        let topics_data = &buf[rel_address_offset + address_header.value_len
+            ..log_header.header_len + log_header.value_len];
+        let topics_rlp = rlp::Rlp::new(topics_data);
+        let topics_header = topics_rlp.payload_info()?;
+        let topic_0_data =
+            &buf[rel_address_offset + address_header.value_len + topics_header.header_len
+                ..log_header.header_len
+                    + address_header.header_len
+                    + address_header.value_len
+                    + topics_header.header_len
+                    + topics_header.value_len];
+        let topic_0_rlp = rlp::Rlp::new(topic_0_data);
+        let topic_0_header = topic_0_rlp.payload_info()?;
+        let rel_sig_offset = log_header.header_len
+            + address_header.header_len
+            + address_header.value_len
+            + topics_header.header_len
+            + topic_0_header.header_len;
+        let event_signature: [u8; 32] = buf[rel_sig_offset..rel_sig_offset + 32].try_into()?;
+        // Each topic takes 33 bytes to encode so we divide this length by 33 to get the number of topics remaining
+        let remaining_topics = buf[rel_sig_offset + topic_0_header.value_len
+            ..log_header.header_len
+                + address_header.header_len
+                + address_header.value_len
+                + topics_header.header_len
+                + topics_header.value_len]
+            .len()
+            / 33;
```

Comment:
Are you 💯  about that ?
Otherwise, following the rlp decoing should work no ? `rlp::Decode_list()` should return you all the topics normally decoded as a `Vec<Vec<u8>>` I _believe_